tests/h/models/groups_test.py

```diff
@@ -54,6 +59,51 @@ def test_type_returns_restricted_for_restricted_groups(factories):
     assert factories.RestrictedGroup().type == 'restricted'
 
 
+def test_it_returns_None_by_default_for_authority_provided_id():
+    group = models.Group(name="abcdefg")
+
+    assert group.authority_provided_id is None
+
+
+def test_it_returns_None_for_groupid_if_authority_provided_id_is_None(factories):
+    group = factories.Group()
+
+    assert group.groupid is None
+
+
+def test_it_returns_formatted_groupid_if_authority_provided_id(factories):
+    group = factories.Group()
+    group.authority_provided_id = 'hithere'
+
+    assert group.groupid == 'group:hithere@{authority}'.format(authority=group.authority)
+
+
+@pytest.mark.parametrize('authority_provided_id', [
+    '%%&whatever',
+    '^flop',
+    '#---',
+    'ßeta',
```

Comment:
Worth adding the empty string as a test case here?

Reply:
Ah, nevermind - I tested this locally and it looks like setting `authority_provided_id` to an empty string has the same effect as setting it to `None`. I presume this equivalence is handled by SQLAlchemy.